### ISSUE TITLE
How to colour <ul><li> bullet points?

### ISSUE THREAD

**Issue by Quirksmode** [jsdf/react-native-htmlview#84]
I have tried this, but it doesn't work:

`ul: {
        color: '#fff'
    },
    li: {
        color: '#fff'
    },`

**Comment by Quirksmode** [comment]
To add to this, the list item text wrap underneath the bullets, is there a way to target these somehow?

**Comment by smhatre59** [comment]
Actually the bullet point is simply a unicode character as defined here:
https://github.com/jsdf/react-native-htmlview/blob/master/htmlToElement.js
Simply go to your npm modules folder and open this file in some editor.Then change the unicode character to something else on line no 11:
const BULLET = '\u2022 ';
Reload your app to see the change in bullet point.
This is not a permanent solution and should be handled in the npm module itself but for now this workaround can help.
I was able to change the bullet point to diamonds by changing the unicode
![screen shot 2017-03-22 at 2 02 42 pm](https://cloud.githubusercontent.com/assets/13968787/24188725/7b6cba4e-0f08-11e7-86d3-4b1f9f7e2020.png)

**Comment by Quirksmode** [comment]
Thanks for getting back to me. How would I change the colour to white and have the size match the font size of the <li> it sits alongside? Also, is there a way to avoid the wrapping issue?

**Comment by smhatre59** [comment]
In that case use the renderNode method to override li tag as follows:
Install react-native-vector-icons
Import them in your project:
import Icon from 'react-native-vector-icons/FontAwesome';
Then define renderNode function as follows:
renderNode(node, index, siblings, parent) {
      if (node.name == 'li') {
        const a = node.attribs;
        return (
          <Text>
          <Icon name="rocket" size={30} color="#900" />
          <Text>asdfsd{"\n"}</Text>
          </Text>
        );
      }
    }

And invoke renderNode in HTMLView as follows:
<HTMLView
        value={htmlContent}
        stylesheet={styles}
        renderNode={(node, index, siblings, parent)=>this.renderNode(node, index, siblings, parent)} 
      />
![screen shot 2017-03-22 at 5 35 32 pm](https://cloud.githubusercontent.com/assets/13968787/24197041/0b58e94e-0f26-11e7-904f-eec73e4810ca.png)
Please remember that li view is nested within Text so don't use View while overriding li tag.
Kindly close this issue if you don't need any more help or revert back on same thread.

**Comment by koenpunt** [comment]
For proper list alignment I found that I have to create a "table" with views.

So something like this:

```jsx
// The function passed to the `renderNode` prop of the `HTMLView`
const customRenderer = (node, index, siblings, parent, domToElement) => {
  if (node.name === 'ol') {
    return (
      <List
        node={node}
        key={`node-${index}`}
        domToElement={domToElement}
      />
    );
  }
  return undefined;
};

// Custom List component
const List = ({ node, bullet, domToElement }) => (
  <View style={styles.list}>
    {node.children.filter(child => child.name === 'li').map((child, index) => (
      <View style={styles.listItem} key={index}>
        <View style={styles.listItemBullet}>
          {index + 1}
        </View>
        <View style={styles.listItemContent}>{domToElement(child.children, child)}</View>
      </View>
    ))}
  </View>
);

const styles = StyleSheet.create({
  listItem: {
    flexDirection: 'row',
    marginBottom: 10,
  },

  listItemBullet: {
    width: 20,
    paddingRight: 10,
  },

  listItemContent: {
    width: 0,
    flexGrow: 1,
  },
});
```

However for this to work you have to use the github master, because `domToElement` is available in the function since 607ab6a805ca9aab448f3a910d610ae53a0708ee, which has not been released yet.

**Comment by Quirksmode** [comment]
Thanks guys, this is really helpful, will give it a try.
